tools/gen-magic-strings.py

```diff
@@ -133,7 +133,8 @@ def process_file(fname):
                 guards = guard_stack[-1]
                 guards[-1] = '!(%s)' % guards[-1]
             elif endif_match is not None:
-                guard_stack.pop()
+                if guard_stack:
+                    guard_stack.pop()
```

Comment:
I don't think that this should be fixed here / this way. If there is an `#endif` in the sources but the stack is empty then we have missed something, which should not go hidden.

I see 2-3 options:
1) Don't fix this.
2) Fix the problem of gen-magic-strings by updating the regexes in lines 122-125 (Right now they cannot recognize preproc directives if they don't start on column 0 of the line. The `'^#` patterns should be updated as `'^ *#`.)
3) Don't fix this in this PR but in a separate one, as mentioned in option 2.

I'd suggest option 3. With check-magic-strings.sh already being fixed. should any error be raised in gen-magic-strings.py, that will not go unnoticed anymore. So the primary objective, which is quite critical, is already solved.

Reply:
For a moment i was confused about the regex change, but then i realized that, the ' is just the opening quote for the regex string.
3, Seems like a good idea. Will make a PR for that in a moment.